mongo/filterstore.go

```diff
@@ -120,130 +136,152 @@ func (s *FilterStore) UpdateFilter(isAuthenticated bool, updatedFilter *models.F
 // GetFilterDimensions returns a list of dimensions from a filter, if the filter is not found an error is returned
 func (s *FilterStore) GetFilterDimensions(filterID string) ([]models.Dimension, error) {
 	session := s.Session.Copy()
+	defer session.Close()
+
 	query := bson.M{"filter_job_id": filterID}
 	dimensionSelect := bson.M{"dimensions": 1}
 	var result models.Filter
-	err := session.DB(Database).C(FiltersCollection).Find(query).Select(dimensionSelect).One(&result)
+
+	err := session.DB(database).C(filtersCollection).Find(query).Select(dimensionSelect).One(&result)
 	if err != nil {
 		if err == mgo.ErrNotFound {
-			return nil, NotFound
+			return nil, errNotFound
 		}
 		return nil, err
 	}
+
 	return result.Dimensions, nil
 }
 
 // GetFilterDimension return a single dimension
 func (s *FilterStore) GetFilterDimension(filterID string, name string) error {
 	session := s.Session.Copy()
+	defer session.Close()
+
 	queryFilter := bson.M{"filter_job_id": filterID}
 	queryDimension := bson.M{"filter_job_id": filterID, "dimensions": bson.M{"$elemMatch": bson.M{"name": name}}}
 	dimensionSelect := bson.M{"dimensions": 1}
 	var result models.Filter
-	err := session.DB(Database).C(FiltersCollection).Find(queryFilter).Select(dimensionSelect).One(&result)
+
+	err := session.DB(database).C(filtersCollection).Find(queryFilter).Select(dimensionSelect).One(&result)
 	if err != nil {
 		if err == mgo.ErrNotFound {
-			return BadRequest
+			return errBadRequest
 		}
 		return err
 	}
-	err = session.DB(Database).C(FiltersCollection).Find(queryDimension).Select(dimensionSelect).One(&result)
+
+	err = session.DB(database).C(filtersCollection).Find(queryDimension).Select(dimensionSelect).One(&result)
 	if err != nil {
 		if err == mgo.ErrNotFound {
-			return DimensionNotFound
+			return errDimensionNotFound
 		}
 		return err
 	}
+
 	return nil
 }
 
 // AddFilterDimension to a filter
 func (s *FilterStore) AddFilterDimension(dimension *models.AddDimension) error {
-	err := s.checkFilterState(dimension.FilterID)
-	if err != nil {
+	if err := s.checkFilterState(dimension.FilterID); err != nil {
 		return err
 	}
 
 	session := s.Session.Copy()
+	defer session.Close()
+
 	queryFilter := bson.M{"filter_job_id": dimension.FilterID}
 	url := fmt.Sprintf("%s/filter/%s/dimensions/%s", s.host, dimension.FilterID, dimension.Name)
 	d := models.Dimension{Name: dimension.Name, Options: dimension.Options, DimensionURL: url}
 	update := bson.M{"$push": bson.M{"dimensions": d}}
-	err = session.DB(Database).C(FiltersCollection).Update(queryFilter, update)
+
+	err := session.DB(database).C(filtersCollection).Update(queryFilter, update)
```

Comment:
As you have done previously, we can reduce lines 199 & 200 to the following 1 liner:

`if err := session.DB(database).C(filtersCollection).Update(queryFilter, update); err != nil {`